transmart-core-db/grails-app/services/org/transmartproject/db/support/InQuery.groovy

```diff
@@ -23,27 +27,42 @@ import static org.transmartproject.db.support.DatabasePortabilityService.Databas
  * it chops list of items and join sub-conditions with OR operations
  * patient_num IN (1, 2, 3, ... 1000) OR patient_num IN (1001, 1002, 1003, ... 1200)
  */
+@CompileStatic
 class InQuery {
 
-    public static final int MAX_LIST_SIZE = 1000
+    public static final int ORACLE_MAX_LIST_SIZE = 1000
+    public static final int POSTGRES_MAX_PARAMETERS = 10000
 
-    @CompileStatic
     static Criterion inValues(String property, Iterable values) {
         if (databaseTypeIsOracle) {
-            def collatedValues = collateValues(values)
+            def collatedValues = collateValues(values, ORACLE_MAX_LIST_SIZE)
             return inCollatedValues(property, collatedValues)
         } else {
             return Restrictions.in(property, values.collect().toArray())
         }
     }
 
+    @CompileDynamic
     static HibernateCriteria addIn(HibernateCriteriaBuilder criteriaBuilder, String property, Iterable values) {
         return criteriaBuilder.add(inValues(property, values))
     }
 
+    static List listIn(HibernateCriteriaBuilder criteriaBuilder, String property, Iterable values) {
```

Comment:
It is not easy to understand what this function does at once. Could you add a doc-string and test that would serve as an example?